### PR TITLE
[VnV Report] [SRS] [VnV Plan] Add FRT Evaluations for Shop Lookup

### LINKED ISSUE #447: [VnV Report] Functional Requirements Evaluation - Shop Lookup
## Content

- Functional Requirements Evaluation

## Rubric Sections

- Content
- Spelling and grammar and style
- VnV Report Matches VnV Plan

### PR DESCRIPTION
**Pull request by TimChoy**
Close #447

### CODE REVIEW

File: docs/SRS/SRS.tex [hunk 1]
```diff
@@ -1437,12 +1438,9 @@ \subsubsection{Shop Lookup}
...
 		      \item Viewpoint: Vehicle Owner
 		            \begin{enumerate}
-			            \item The system shall allow the user to enter a distance range
-			            \item The system shall allow the user to enter a postal code
-			            \item The system shall allow the user to enter a shop name
-			            \item The system shall display all auto repair shops within the distance provided, or any shops at a
-			                  postal code, or any shops which match the shop's name within the specified distance
-			            \item The system shall show the user's distance to each shop provided
+			            \item The system shall allow the user to enter a shop name or service name
+			            \item The system shall allow the user to filter by part types used in the service
```
Comment (LeonSo7) [inline docs/SRS/SRS.tex]:
```suggestion
			            \item The system shall allow the user to filter by part type and condition used in the service
```

File: docs/VnVPlan/VnVPlan.tex [hunk 2]
```diff
@@ -1166,10 +1167,10 @@ \subsubsection{Shop Lookup}
 
 	      Initial State: Shop Owner/Employee Account with a registered shop, Customer Account exists
 
-	      Input: The Customer Account attempts to view a list of shops, using the parameters of distance,
-	      postal code, and/or shop name
+	      Input: The Customer Account attempts to view a list of shops, using the parameters of shop name,
+	      service name, and part types
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      service name, and part types and conditions
```

File: docs/VnVPlan/VnVPlan.tex [hunk 3]
```diff
@@ -1166,10 +1167,10 @@ \subsubsection{Shop Lookup}
 
 	      Initial State: Shop Owner/Employee Account with a registered shop, Customer Account exists
 
-	      Input: The Customer Account attempts to view a list of shops, using the parameters of distance,
-	      postal code, and/or shop name
+	      Input: The Customer Account attempts to view a list of shops, using the parameters of shop name,
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      Input: The Customer Account attempts to view a list of shops using the parameters of shop name,
```

File: docs/VnVReport/VnVReport.tex [hunk 4]
```diff
@@ -622,6 +623,29 @@ \subsection{Services}
...
+\subsection{Shop Lookup}
+
+\begin{enumerate}
+	\item \textbf{FRT-BE32-1}
+
+	      Control: Manual
+
+	      Initial State: Shop Owner/Employee Account with a registered shop, Customer account exists
+
+	      Input: The Customer account attempts to view a list of shops, using the parameters of shop name,
```
Comment (LeonSo7) [inline docs/VnVReport/VnVReport.tex]:
```suggestion
	      Input: The Customer account attempts to view a list of shops using the parameters of shop name,
```

**Review by LeonSo7 — COMMENTED** [review]
You might also want a test for unexpected inputs (e.g., from looking at your api, invalid input should result in 400 status), or cases where no shop exists that matches the filter (no shops should be returned)

**Comment by TimChoy** [comment]
> You might also want a test for unexpected inputs (e.g., from looking at your api, invalid input should result in 400 status)

I'm not sure how to test for that since the front end forces the input to be a string and the api is there as a "just in case" precaution.

> or cases where no shop exists that matches the filter (no shops should be returned)

This is technically covered in the description of the FRT evaluation. 

Also wouldn't these fit better in the unit test section rather than the functional requirements system tests?

**Review by LeonSo7 — APPROVED** [review]

**Comment by LeonSo7** [comment]
> Also wouldn't these fit better in the unit test section rather than the functional requirements system tests?

Ya add it to unit tests

**Pull request merged**